pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/TNCudaBase.hpp

```diff
@@ -141,6 +155,128 @@ class TNCudaBase : public TensornetBase<Precision, Derived> {
+        }
+    }
+
+    /**
+     * @brief Append multiple gate tensors to the compute graph.
+     * NOTE: This function does not update the quantum state but only appends
+     * gate tensor operator to the graph.
+     * @param ops Vector of gate names to be applied in order.
+     * @param ops_wires Vector of wires on which to apply index-matched gate
+     * name.
+     * @param ops_adjoint Indicates whether gate at matched index is to be
+     * inverted.
+     */
+    void applyOperations(const std::vector<std::string> &ops,
+                         const std::vector<std::vector<std::size_t>> &ops_wires,
+                         const std::vector<bool> &ops_adjoint) {
+        const std::size_t numOperations = ops.size();
+        PL_ABORT_IF_NOT(
+            numOperations == ops_wires.size(),
+            "Invalid arguments: number of operations, wires, and inverses "
+            "must all be equal");
+        PL_ABORT_IF_NOT(
+            numOperations == ops_adjoint.size(),
+            "Invalid arguments: number of operations, wires and inverses"
+            "must all be equal");
+        for (std::size_t i = 0; i < numOperations; i++) {
+            this->applyOperation(ops[i], ops_wires[i], ops_adjoint[i], {});
+        }
+    }
+
+    /**
+     * @brief Append a single gate tensor to the compute graph.
+     * NOTE: This function does not update the quantum state but only appends
+     * gate tensor operator to the graph.
+     * @param opName Gate's name.
+     * @param wires Wires to apply gate to.
+     * @param adjoint Indicates whether to use adjoint of gate.
+     * @param params Optional parameter list for parametric gates.
+     * @param gate_matrix Optional gate matrix for custom gates.
+     */
+    void applyOperation(const std::string &opName,
+                        const std::vector<std::size_t> &wires,
+                        bool adjoint = false,
+                        const std::vector<PrecisionT> &params = {0.0},
+                        const std::vector<ComplexT> &gate_matrix = {}) {
+        auto &&par = (params.empty()) ? std::vector<PrecisionT>{0.0} : params;
+        DataBuffer<PrecisionT, int> dummy_device_data(
+            Pennylane::Util::exp2(wires.size()), getDevTag());
+        int64_t id;
+        std::vector<int32_t> stateModes(wires.size());
+        std::transform(
+            wires.begin(), wires.end(), stateModes.begin(), [&](std::size_t x) {
+                return static_cast<int32_t>(BaseType::getNumQubits() - 1 - x);
+            });
+
+        // TODO: Need changes to support to the controlled gate tensor API once
+        // the API is finalized in cutensornet lib.
+        //  Note `adjoint` in the cutensornet context indicates whether or not
+        //  all tensor elements of the tensor operator will be complex
+        //  conjugated. `adjoint` in the following API is not equivalent to
+        //  `inverse` in the lightning context
```

Comment:
Minor, feel free to fix this in your followup PR,
```suggestion
        // Note `adjoint` in the cutensornet context indicates whether or not
        // all tensor elements of the tensor operator will be complex
        // conjugated. `adjoint` in the following API is not equivalent to
        // `inverse` in the lightning context
```